File: 33-Search-in-Rotated-Sorted-Array/solve.c
class Solution {
public:
    int search(vector<int>& nums, int target) {
        
        if (nums.size() == 0)
            return -1;
        int begin = 0;
        int end = nums.size() -1;
        int mid = -1;
        while (begin <= end){
            mid = (begin + end) / 2;
            if (nums[mid] == target)
                return mid;
            if (nums[begin] <= nums[mid]){
                if ( target < nums[mid] && target >= nums[begin] )
                    end = mid -1;
                else
                    begin = mid + 1;
            }
            else{
                if( target > nums[mid] && target <= nums[end])
                    begin = mid + 1;
                else
                    end = mid - 1;
            }
        }
        
       return -1;
        
        
    }
};

File: 1-Two-Sum/solve.java
import java.util.*;
public class Solution {
    public int[] twoSum(int[] nums, int target) {
        
		int[] result = new int[2];

		Hashtable<Integer, Integer> hashtable = new Hashtable<Integer, Integer>();

		for (int i = 0; i < nums.length; i++) {

			if (hashtable.containsKey(target - nums[i])) {
				result[0] = hashtable.get(target - nums[i]) + 1;
				result[1] = i + 1;
			}
			hashtable.put(nums[i], i);

		}
		return result;
    }
}
